testsCommon/p2p/messengerMock.go

```diff
@@ -0,0 +1,84 @@
+package p2p
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+)
+
+// MessengerMock -
+type MessengerMock struct {
```

Comment:
maybe add a lock on the struct and use the lock in the relevant methods, so when used with different goroutines in an integration tests it does not fail with race conditions

Reply:
done. Refactored the mock